cpp/src/arrow/util/decimal_test.cc

```diff
@@ -802,34 +814,21 @@ class TestDecimalFromReal : public ::testing::Test {
     const std::vector<ParamType> params{
         // clang-format off
         {0.0f, 1, 0, "0"},
-        {-0.0f, 1, 0, "0"},
```

Comment:
Why remove these cases?  We can still convert negative numbers right?  Isn't it just less precise?

Reply:
Because `CheckDecimalFromReal` and `CheckDecimalFromRealIntegerString` now automatically deduce the negative test cases from the positive ones.